File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def to_binary(num_str):
    num = int(num_str)
    result = ""
    while num != 1:
        result = str(num % 2) + result
        num = num//2
    result = str(num) + result
    return result
def solution(s):
    cnt_zero = 0
    cnt_transform = 0
    result = s
    # 0 카운팅 및 제거
    # 길이 저장 및 이진수 변환
    # "1"이 될 때 까지 반복
    while result != "1":
        cnt_zero += result.count("0")
        result = result.replace("0","")
        #print(result)
        result = to_binary(len(result))
        cnt_transform += 1
    return ([cnt_transform, cnt_zero])

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;

class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public static String intToBinary(int num){
        // 2 mod 연산 값 구하기
        // 2 나누기
        // 몫이 2보다 작으면 끝 
        
        StringBuilder b = new StringBuilder();
        while(num >= 2){
            int remain = num % 2;
            b.append(remain);
            num = num / 2;
        }
        b.append(num);
        b.reverse();
        return b.toString();
    }
    public int[] solution(String s) {
        /*
        한사이클 마다 
        1. 0 개수 구하기, result에 저장
        2. 전체 길이에서 0 개수 빼기
        3. 구한 값 이진수 변환하기
        
        --> 전체 길이가 1이 될 때 까지 반복
        */
        int cnt = 0;
        int total = 0;
        while(s.length() >= 2){
            cnt++;
            int zero = 0;
            for(int i=0; i<s.length(); i++){
                if(s.charAt(i) == '0'){
                    zero++;
                }
            }
            //log(zero);
            int temp = s.length() - zero;
            total += zero; 
            s = intToBinary(temp);
        }
        return new int[]{cnt, total};
    }
}